pkg/common/configs/configvalidator.go

```diff
@@ -336,14 +348,30 @@ func checkLimit(limit Limit, currIdx int, userWildCardIdx, groupWildCardIdx *int
 		// It means the wildcard for group should be the last item for limits object list which including the group name,
 		// and we should only set one wildcard group for all limits
 		if name == "*" {
-			if *groupWildCardIdx != -1 && currIdx > *groupWildCardIdx {
+			if limitCheckHelper.groupWildCardIdx != -1 && currIdx > limitCheckHelper.groupWildCardIdx {
 				return fmt.Errorf("should not set more than one wildcard group")
 			}
-			*groupWildCardIdx = currIdx
-		} else if *groupWildCardIdx != -1 && currIdx > *groupWildCardIdx {
+			limitCheckHelper.groupWildCardIdx = currIdx
+		} else if limitCheckHelper.groupWildCardIdx != -1 && currIdx > limitCheckHelper.groupWildCardIdx {
 			return fmt.Errorf("should not set no wildcard group %s after wildcard group limit", name)
+		} else {
+			limitCheckHelper.groupSpecificIdx = currIdx
+		}
+		if limitCheckHelper.existedGroupName[name] {
+			return fmt.Errorf("duplicated group name %s , already existed", name)
 		}
+		limitCheckHelper.existedGroupName[name] = true
 	}
+
+	// Specifying a wildcard for the group limit sets a cumulative limit for all users in that queue.
+	// If there is no specific group mentioned the wildcard group limit would thus be the same as the queue limit.
+	// For that reason we do not allow specifying only one group limit that is using the wildcard.
+	// There must be at least one limit with a group name defined.
+	if limitCheckHelper.groupWildCardIdx != -1 && limitCheckHelper.groupSpecificIdx == -1 {
+		return fmt.Errorf("should not specify only one group limit that is using the wildcard. " +
+			"There must be at least one limit with a group name defined ")
+	}
+
```

Comment:
Do we really need groupWildCardIdx? Can we do something like this?

if existedGroupName has *  and len of existedGroupName is 1 then return error.

Reply:
Good suggestion, addressed in latest PR.